packages/fast-components/src/design-system-provider/design-system-provider.ts

```diff
@@ -1,98 +1,271 @@
+            // If a property on *this* object that is declared as a design system property
+            this.designSystem[key] = this[key];
+            const desginSystemProperty = this.designSystemProperties[key];
+
+            if (desginSystemProperty && desginSystemProperty.customProperty) {
+                this.setCustomProperty({
+                    name: desginSystemProperty.customPropertyName,
+                    value: this[key],
+                });
+            }
+            this.consumers.forEach(this.writeConsumerRecipeData);
+        } else if (source === this.provider && key === designSystemKey) {
+            // If our provider's design system has changed
+            this.syncDesignSystemWithProvider();
+            this.consumers.forEach(this.writeConsumerRecipeData);
         }
     }
 
-    @designSystemProperty
-    public backgroundColor: string = "#FFFFFF";
+    public suscribe(consumer: DesignSystemConsumer): void {
+        if (!this.consumers.has(consumer)) {
+            this.consumers.add(consumer);
+            this.writeConsumerRecipeData(consumer);
+        }
+    }
 
-    @designSystemProperty
-    public baseHeightMultiplier: number = 8;
+    public unsubscribe(consumer: DesignSystemConsumer): void {
+        this.consumers.delete(consumer);
+    }
 
-    @designSystemProperty
-    public bodyFont: string = "Segoe UI, sans-serif";
+    private writeConsumerRecipeData = (consumer: DesignSystemConsumer) => {
+        consumer.recipes.forEach(recipe => {
+            this.setCustomProperty({
+                name: recipe.name,
+                // use spread on the designSystem object to circumvent memoization
+                // done in the color recipes - we use the same *reference* in WC
+                // for performance improvements but that throws off the recipes
+                // We should look at making the recipes use simple args that
+                // we can individuall memoize.
+                value: recipe.resolver.bind(this, { ...this.designSystem }),
+            });
+        });
+    };
 
-    @designSystemProperty
-    public cornerRadius: number = 2;
+    /**
+     * Syncronizes the provider's design system with the local
+     * overrides. Any value defined on the instance will take priority
+     * over the value defined by the provider
+     */
+    private syncDesignSystemWithProvider(): void {
+        if (this.provider) {
+            Object.keys(this.provider.designSystem).forEach(key => {
+                if (this[key] === void 0) {
+                    this.designSystem[key] = this.provider!.designSystem[key];
+                }
+            });
+        }
+    }
 
-    @designSystemProperty
-    public disabledOpacity = 0.3;
+    /**
+     * Invoked when the provider observable property defined by the consumer is changed
+     * @param prev the previous value
+     * @param next the next value
+     */
+    private providerChanged(

```

Comment:
This feels a bit wonky to me because it is actually the *consumer* that defines the observable property. If we were to add a providerChanged callback to the consumer then this wouldn't fire. We need the prev/next values though to unsubscribe. Happy to hear thoughts on an alternative way to approach this.

Reply:
That might not be so strange, I could see it as the consumer having to define the recipes it is interested in consuming.